relay-server/src/actors/project_redis.rs

```diff
@@ -52,8 +54,17 @@ impl RedisProjectSource {
             .map_err(RedisError::Redis)?;
 
         let raw_response = match raw_response_opt {
-            Some(response) => response,
-            None => return Ok(None),
+            Some(response) => {
+                metric!(counter(RelayCounters::ProjectStateRedis) += 1, hit = "true");
+                response
+            }
+            None => {
+                metric!(
+                    counter(RelayCounters::ProjectStateRedis) += 1,
+                    hit = "false"
+                );
```

Comment:
I was wondering about the impact of the sampling here. Can it be that metrics differ enough from reality to not provide meaningful data? I don't know if it's worth the effort, but here's a suggestion: store the counters in memory and flush them every n times with sampling rate set to 100%. Besides having accurate data (except for the last bucket), this approach also generates less overhead generated by metrics, an issue we've had not so long ago: https://github.com/getsentry/relay/pull/1316.

Reply:
I think that's correct but i feel like we should "fix" the cadence crate somehow to do this aggregation for us

Reply:
oh my, i missed this comment.  but basically this has been one of my major issues with statsd from day 1.  what you describe is much closer to what prometheus does and what I'd also like to do.  I even started a hackweek project the first year to work on that.  so i'm like 1000% for this.